src/main/java/edu/harvard/iq/dataverse/util/bagit/BagGenerator.java

```diff
@@ -115,6 +115,7 @@ public class BagGenerator {
     private boolean usetemp = false;
 
     private int numConnections = 8;
+    public static final String BAG_GENERATOR_THREADS = ":BagGeneratorThreads";

```

Comment:
Instead of having a String here, can we keep all the database settings centralized in the `Key` enum at SettingsServiceBean?

I tried this locally and the following seems to work:

`public static final String BAG_GENERATOR_THREADS = SettingsServiceBean.Key.BagGeneratorThreads.toString();`

(I had to import SettingsServiceBean.)

I see that DuraCloudSubmitToArchiveCommand also has strings like :DuraCloudPort, :DuraCloudHost, and :DuraCloudContext hard-coded, but we could centralize those later.

Reply:
I've purposely avoided that so far with the hope that at some point these archivers will be packaged separately. Right now they are loaded via reflection (you specify the classname in a property) and there should be no references to the specific Archiver classes or their properties in the main codebase. (As you note though we do have their properties in the master list in the guides so far.)
( I hope it isn't too much work for these classes and their dependencies to be pulled out into separate jars but it isn't something I know how to do off-hand - perhaps @poikilotherm?)